File: 02 tests/ex6_matrix.js
// run with: deno run ex6_matrix.js

/* ex6_matrix.js

Matrix multiplication

:author:  wolf
:created: 2025.05.22
*/

const iterations = 3;
const timing_results = [];

// define the test function
// --- function to be timed-- -

function jsMatrixMultiply(n) {
//    const A = Array(n).fill().map(() => Array(n).fill().map(() => Math.floor(Math.random() * 101)));
    const A = Array(n).fill().map(() => Array(n).fill().map(() => Math.floor(Math.random() * 101)));
    const B = Array(n).fill().map(() => Array(n).fill().map(() => Math.floor(Math.random() * 101)));
    const result = Array(n).fill().map(() => Array(n).fill(0));
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            for (let k = 0; k < n; k++) {
                result[i][j] += A[i][k] * B[k][j];
            }
        }
    }
    return result;
}


for (let i = 0; i < iterations; i++) {
    const start = performance.now();

    // call the test function
    jsMatrixMultiply(10**3);

    const end = performance.now();

    timing_results.push(end - start);
}


console.log("Elapsed time: ",
    timing_results.reduce((a, b) => a + b) / iterations / 1000, "seconds");

File: 02 tests/ex6_matrix.py
#!/usr/bin/env python3

""" ex6_matrix.py

Matrix multiplication

:author:	wolf
:created:	2025.05.22
"""

import time
import random

iterations = 3
timing_results = []

# define the test function
# --- function to be timed ---


def python_matrix_multiply(n):
    A = [[random.randint(0, 100) for _ in range(n)] for _ in range(n)]
    B = [[random.randint(0, 100) for _ in range(n)] for _ in range(n)]
    result = [[0] * n for _ in range(n)]
    for i in range(n):
        for j in range(n):
            for k in range(n):
                result[i][j] += A[i][k] * B[k][j]
    return result


for i in range(iterations):
    start = time.perf_counter()

    # call the test function
    python_matrix_multiply(10**3)

    end = time.perf_counter()

    timing_results.append(end - start)


print(f"Elapsed time:  {sum(timing_results)/iterations:.4f} seconds")
